doc/RELEASE_WALKTHROUGH.rst

```diff
@@ -154,100 +149,71 @@ Sanity check::
-
-If some wheel builds fail for unrelated reasons, you can re-run them:
-
-- On GitHub actions select `Wheel builder`_ click on the task that contains
-  the build you want to re-run, it will have the tag as the branch. On the
-  upper right will be a re-run button, hit it and select "re-run failed"
-
-If some wheels fail to upload to anaconda, you can select those builds in the
-`Wheel builder`_ and manually download the build artifact. This is a temporary
-workaround, but sometimes the quickest way to get a release out.
-
-.. _`staging repository`: https://anaconda.org/multibuild-wheels-staging/numpy/files
-.. _`Wheel builder`: https://github.com/numpy/numpy/actions/workflows/wheels.yml
-
-
-3. Download wheels
-------------------
+   $ git tag -d v2.4.0
+   $ git push --delete upstream v2.4.0
 
-When the wheels have all been successfully built and staged, download them from the
-Anaconda staging directory using the ``tools/download-wheels.py`` script::
 
-    $ cd ../numpy
-    $ mkdir -p release/installers
-    $ python3 tools/download-wheels.py 2.1.0
-
-
-4. Generate the README files
-----------------------------
-
-This needs to be done after all installers are downloaded, but before the pavement
-file is updated for continued development::
-
-    $ python write_release 2.1.0
+2. Build wheels and sdist
+-------------------------
 
+Create a ``maintenance/2.4.x`` branch in the ``numpy-release`` repository,
+and open a PR changing the ``SOURCE_REF_TO_BUILD`` identifier at the top of
+``.github/workflows/wheels.yml`` to ``v2.4.0``. That will do a full set of
+wheel builds on the PR, if everything looks good merge the PR.
 
-5. Upload to PyPI
------------------
+All wheels are currently built in that repository on GitHub Actions, they take
+about 1 hour to build. 
 
-Upload to PyPI using ``twine``::
+If you wish to manually trigger a wheel build, you can do so: in your browser,
+go to `numpy-release/actions/workflows/wheels.yml <https://github.com/numpy/numpy-release/actions/workflows/wheels.yml>`__
+and click on the "Run workflow" button, then choose the tag to build. If some
+wheel builds fail for unrelated reasons, you can re-run them as normal
+in the GitHub Actions UI with "re-run failed".
 
-    $ cd ../numpy
-    $ twine upload release/installers/*.whl
-    $ twine upload release/installers/*.gz  # Upload last.
+Once you are ready to publish a release to PyPI, use that same "Run workflow"
+button and choose ``pypi`` in the *environment* dropdown. All wheels and the
+sdist will build and be ready to release to PyPI after manual inspection that
+everything passed. E.g., the number of artifacts is correct, and the wheel
+filenames and sizes look as expected. If desired, you can also download an
+artifact for local unzipping and inspection. You will get an email notification
+as well with a "Review pending deployments" link. Once you're ready, press the
+button to start the uploads to PyPI, which will complete the release.
```

Comment:
Which button?

Reply:
I can't recreate the popup with the exact text right now without running a new deployment, but it's something like "Approve deployment". There's only one button that shows up once you follow "Review pending deployments".

How about I tweak the text to say "Once you're done reviewing, press the "Approve deployment" button to start ..." and then if it's not 100% accurate, you can make it more accurate when you use these instructions for 2.4.0?